### PR TITLE
Refactor payment/history.js

### LINKED ISSUE #8037: Refactor payment/history.js
## Test plan 

https://github.com/brave/browser-laptop/pull/8038#issue-218756988

----

**Describe the issue you encountered:**
Refactor payment/history.js

The payment history should be styled like this based on https://github.com/brave/browser-laptop/pull/6047#issue-193827878:

> screenshot 2016-12-15 3 36 37

The original mockup provided by @bradleyrichter is available here: https://github.com/brave/browser-laptop/issues/6202

Currently:
screenshot 2017-04-02 18 13 47

- Any related issues: #7381 #7423

### PR DESCRIPTION
**Pull request by luixxiul**
See: cropped picture of the 6th commit on https://github.com/brave/browser-laptop/pull/6047#issue-193827878

Closes #8037

- Changed `aphrodite` to `aphrodite/no-important` (to speed up removing the LESS files)
  Aphrodite without no-important overwrites the existing code, which should have been removed from the LESS files at the same time. It will make it difficult to know which styles should be kept and removed, having confidence in that changes would not cause regressions. It might be slowing down the refactoring work.

- Modified modalOverlay.js by adding `customHeaderClassesStr`
  The intent of the change is to make it possible to apply custom styles to `sectionTitle`. In this case, `braveMediumOrange` to the title.

- Introduced `paymentStylesVariables` and `paymentHistoryTablePadding`
- Appied it to `leftRow` and `paymentHistoryOverlay__title`
  It aligns the padding-left of `sectionTitle` and the most left row.

- Set `position:sticky` to the table header
  It always display the table header. `Sticky` has been available on chromium since a couple of versions.

- Set `columnHeight` and `columnPadding`

- Reverted `commonStyles.primaryButton` to `primaryButton` temporarily
  There have been already `browserButton` and `primaryButton` available with commonStyles.js, still these styles are overwritten by ones which button.less defines. We have to work on refactoring button.less at first.

Also:
- Removed medium
- Removed pull-left, which is no longer used

Auditors:

Test Plan:
1. Open about:preferences#payments
2. The modal dialog looks like the cropped picture of the 6th commit on https://github.com/brave/browser-laptop/pull/6047#issue-193827878

- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).

Test Plan:

### CODE REVIEW

**Comment by luixxiul** [comment]
https://github.com/brave/browser-laptop/pull/6047#issue-193827878

screenshot 2016-12-15 3 36 37

Currently:

screenshot 2017-04-02 18 13 47

history1

With this PR:

screenshot 2017-04-02 19 32 37

history2

File: app/renderer/components/preferences/payment/history.js [hunk 1]
```diff
@@ -113,82 +113,103 @@ class HistoryFooter extends ImmutableComponent {
...
-    return <div className={css(styles.historyFooter)}>
-      <div className={css(styles.nextPayment)}>
+    return <div className={css(styles.flexAlignCenter, styles.historyFooter)}>
+      <div className={css(styles.historyFooter__nextPayment)}>
         <span data-l10n-id={l10nDataId} data-l10n-args={JSON.stringify(l10nDataArgs)} />
       </div>
+      {/* TODO: refactor button.less */}
       <Button l10nId='paymentHistoryOKText'
-        className={css(commonStyles.primaryButton)}
+        className='primaryButton'
```
Comment (NejcZdovc) [inline app/renderer/components/preferences/payment/history.js]: Why did you change this back to regular class?

Comment (luixxiul) [inline app/renderer/components/preferences/payment/history.js]: As I wrote in the commit message:

> There have been already browserButton and primaryButton available with commonStyles.js, still these styles are overwritten by ones which button.less defines. We have to work on refactoring button.less at first.

It is because `aphrodite` was changed to `aphrodite/no-important`. I left the reasoning in the commit message on this change too.

Comment (NejcZdovc) [inline app/renderer/components/preferences/payment/history.js]: Sorry, I didn't read you comment in the commit. Make sense, so let's create an issue for this and track all things that need's to be done there.

Comment (luixxiul) [inline app/renderer/components/preferences/payment/history.js]: +1!!

File: app/renderer/components/preferences/paymentsTab.js [hunk 2]
```diff
@@ -19,10 +20,14 @@ const {LedgerBackupContent, LedgerBackupFooter} = require('./payment/ledgerBacku
 const {LedgerRecoveryContent, LedgerRecoveryFooter} = require('./payment/ledgerRecovery')
 
 // style
-const cx = require('../../../../js/lib/classSet')
 const globalStyles = require('../styles/global')
+<<<<<<< HEAD
```
Comment (cezaraugusto) [inline app/renderer/components/preferences/paymentsTab.js]: I think we don't want it ;)

File: app/renderer/components/preferences/paymentsTab.js [hunk 3]
```diff
@@ -19,10 +20,14 @@ const {LedgerBackupContent, LedgerBackupFooter} = require('./payment/ledgerBacku
...
 const {paymentStyles} = require('../styles/payment')
 const settingIcon = require('../../../extensions/brave/img/ledger/icon_settings.svg')
+=======
```
Comment (cezaraugusto) [inline app/renderer/components/preferences/paymentsTab.js]: (2)

File: app/renderer/components/preferences/paymentsTab.js [hunk 4]
```diff
@@ -19,10 +20,14 @@ const {LedgerBackupContent, LedgerBackupFooter} = require('./payment/ledgerBacku
...
+const {paymentStyles, paymentStylesVariables} = require('../styles/payment')
+const advanceIcon = require('../../../extensions/brave/img/ledger/icon_settings.svg')
+>>>>>>> Refactor payment/history.js
```
Comment (cezaraugusto) [inline app/renderer/components/preferences/paymentsTab.js]: (3)

File: app/renderer/components/preferences/paymentsTab.js [hunk 5]
```diff
@@ -364,6 +385,21 @@ const styles = StyleSheet.create({
     ':hover': {
       backgroundColor: globalStyles.color.chromeTertiary
     }
+<<<<<<< HEAD
```
Comment (cezaraugusto) [inline app/renderer/components/preferences/paymentsTab.js]: :)

File: app/renderer/components/preferences/paymentsTab.js [hunk 6]
```diff
@@ -160,8 +166,12 @@ class PaymentsTab extends ImmutableComponent {
         />
         : null
       }
+<<<<<<< HEAD
```
Comment (cezaraugusto) [inline app/renderer/components/preferences/paymentsTab.js]: same

File: app/renderer/components/preferences/paymentsTab.js [hunk 7]
```diff
@@ -276,12 +286,23 @@ const styles = StyleSheet.create({
   switchWrap: {
     width: paymentStyles.width.tableCell
   },
+<<<<<<< HEAD
```
Comment (cezaraugusto) [inline app/renderer/components/preferences/paymentsTab.js]: (._.)

File: app/renderer/components/preferences/paymentsTab.js [hunk 8]
```diff
@@ -276,12 +286,23 @@ const styles = StyleSheet.create({
...
   switchWrap__switchControl: {
     // TODO: Refactor switchControls.less
     paddingTop: '0 !important',
     paddingBottom: '0 !important'
   },
   switchWrap__switchSpan: {
+=======
```
Comment (cezaraugusto) [inline app/renderer/components/preferences/paymentsTab.js]: ( .-. )

File: app/renderer/components/preferences/paymentsTab.js [hunk 9]
```diff
@@ -276,12 +286,23 @@ const styles = StyleSheet.create({
...
+
+  // TODO: refactor switchControls.less
+  switchControl: {
+    paddingTop: 0,
+    paddingBottom: 0
+  },
+
+  switchSpan: {
+>>>>>>> Refactor payment/history.js
```
Comment (cezaraugusto) [inline app/renderer/components/preferences/paymentsTab.js]: (._.)

File: app/renderer/components/preferences/paymentsTab.js [hunk 10]
```diff
@@ -364,6 +385,21 @@ const styles = StyleSheet.create({
...
+  },
+
+  paymentHistoryOverlay__title: {
+    // TODO: refactor preferences.less to remove !important
+    // See: .paymentsContainer .modal .dialog .dialog-header .sectionTitle and .modal .dialog .sectionTitle
+
+    color: `${globalStyles.color.braveMediumOrange} !important`,
+    padding: `25px 0 25px ${paymentStylesVariables.spacing.paymentHistoryTablePadding} !important`,
+    textIndent: '0 !important'
+>>>>>>> Refactor payment/history.js
```
Comment (cezaraugusto) [inline app/renderer/components/preferences/paymentsTab.js]: (｡◕‿‿◕｡)

File: app/renderer/components/preferences/paymentsTab.js [hunk 11]
```diff
@@ -160,8 +166,12 @@ class PaymentsTab extends ImmutableComponent {
         />
         : null
       }
+<<<<<<< HEAD
 
       <div className={css(styles.flexAlignEnd)}>
+=======
+      <div className={css(styles.titleBar)}>
+>>>>>>> Refactor payment/history.js
```
Comment (cezaraugusto) [inline app/renderer/components/preferences/paymentsTab.js]: (｡◕‿◕｡)

**Comment by luixxiul** [comment]
@cezaraugusto Wow, maybe they were added during rebase. Will fix soon!

File: app/renderer/components/preferences/payment/history.js [hunk 12]
```diff
@@ -113,82 +113,103 @@ class HistoryFooter extends ImmutableComponent {
...
 
   headerContainer: {
-    borderBottom: `2px solid ${globalStyles.color.lightGray}`
+    paddingTop: columnPadding,
+    paddingBottom: columnPadding,
+    borderBottom: `2px solid ${globalStyles.color.lightGray}`,
+    height: columnHeight,
+    alignItems: 'center',
+    cursor: 'default',
+    WebkitUserSelect: 'none'
```
Comment (NejcZdovc) [inline app/renderer/components/preferences/payment/history.js]: let's use basic user select and not a webkit

Comment (luixxiul) [inline app/renderer/components/preferences/payment/history.js]: Updated: https://github.com/brave/browser-laptop/pull/8038/commits/ae31c149379815d209f12e1ac7e4042c79658d74#diff-d7a7d22ddb5b69ebc41381e4864ff9d0R166

**Comment by luixxiul** [comment]
pinging @bradleyrichter for design review :-)

**Comment by luixxiul** [comment]
also cc @NejcZdovc for review

**Comment by luixxiul** [comment]
the base branch was changed as it is not related with commonForm and @NejcZdovc would work on something which is depentent on this change.

**Comment by luixxiul** [comment]
@NejcZdovc bumped to 0.14.3

**Comment by luixxiul** [comment]
@bsclifton @cezaraugusto I added both of you as a reviewer. would you please review this one?

**Comment by luixxiul** [comment]
@cezaraugusto thanks for reviewing.

**Comment by luixxiul** [comment]
@bsclifton please change the milestone if this is merged to 0.14.2, thanks.

**Comment by luixxiul** [comment]
FYI: there are several reports that `position:sticky` does not work on chromium 58. See https://developers.google.com/web/updates/2016/12/position-sticky for the comments.
